lib/client/api.go

```diff
@@ -3438,7 +3438,7 @@ func (tc *TeleportClient) getSSHLoginFunc(pr *webclient.PingResponse) (SSHLoginF
 		}
 		return tc.pwdlessLogin, nil
 	case authType == constants.Local:
-		if pr.Auth.AllowPasswordless && tc.checkIfPwdlessRegisteredWithoutInteraction(pr) {
+		if tc.canDefaultToPasswordless(pr) {
```

Comment:
I would still make this a separate case, I find it weird to have branches inside a switch like this.